pennylane/ops/op_math/exp.py

```diff
@@ -284,6 +284,42 @@ def _recursive_decomposition(self, base: Operator, coeff: complex):
             coeffs = [c * coeff for c in coeffs]
             return self._trotter_decomposition(ops, coeffs)
 
+        with qml.queuing.AnnotatedQueue() as q:
+
+            qml.cond(
+                qml.math.isclose(qml.math.real(coeff), 0),
+                lambda: self._smart_decomposition(coeff, base),
+                lambda: self._raise_decomposition_undefined_error(coeff, base),
+            )()
+
+        return q.queue
+
+    def _raise_decomposition_undefined_error(self, *args):
+        """Raise an error when the decomposition is not defined."""
+
+        if qml.math.is_abstract(args[0]):
+            # Do not raise an error during tracing
+            return
```

Comment:
@astralcai this is a relatively new feature, but we now support the ability to raise runtime assertions with qjit: https://docs.pennylane.ai/projects/catalyst/en/latest/code/api/catalyst.debug_assert.html

Reply:
of course, this would only work if https://docs.pennylane.ai/en/stable/code/api/pennylane.compiler.active.html

Reply:
nope, unfortunately not yet